### PR TITLE
MULE-8679, fix the way the file path is obtained

### PR DESCRIPTION
**Pull request by damiansima**

### CODE REVIEW

File: core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java [hunk 1]
```diff
@@ -63,23 +64,27 @@ public static void clearLoggingConfig()
...
+        File folder = new File(folderPath);
+
         if (folder != null && "target".equals(folder.getParentFile().getName()))
         {
             folder = folder.getParentFile();
         }
 
         File logConfigFile = FileUtils.findFileByName(folder, "log4j2-test.xml", true);
-        if (logConfigFile == null)
+        if (logConfigFile == null )
```
Comment (marianogonzalez) [inline core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java]: remove empty space

File: core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java [hunk 2]
```diff
@@ -63,23 +64,27 @@ public static void clearLoggingConfig()
...
             logConfigFile = FileUtils.findFileByName(folder, "log4j2.xml", true);
         }
 
-        String logConfigPath = logConfigFile != null ? logConfigFile.getAbsolutePath() : NULL_CONFIG_FILE;
+        String logConfigPath = logConfigFile != null && !logConfigFile.isDirectory() ? logConfigFile.getAbsolutePath() : NULL_CONFIG_FILE;
         LOGGING_CONFIG_CACHE.set(logConfigPath);
         return logConfigPath;
     }
 
+
```
Comment (marianogonzalez) [inline core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java]: remove empty line

File: core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java [hunk 3]
```diff
@@ -63,23 +64,27 @@ public static void clearLoggingConfig()
 
     private static String findLogConfigurationPath(Class<?> testClass)
     {
-        File folder = new File(testClass.getClassLoader().getResource("").getPath().toString());
+        String folderPath = testClass.getClassLoader().getResource("").getPath().toString();
+        folderPath = URLDecoder.decode(folderPath);
```
Comment (marianogonzalez) [inline core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java]: don;t use deprecated methods

File: core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java (hunk 1 above)
Comment (marcosnc) [inline core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java]: remove space

File: core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java (hunk 3 above)
Comment (marcosnc) [inline core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java]: Just add URLDecoder.decode call in the existing line.

File: core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java [hunk 4]
```diff
@@ -63,23 +64,27 @@ public static void clearLoggingConfig()
...
 
         File logConfigFile = FileUtils.findFileByName(folder, "log4j2-test.xml", true);
-        if (logConfigFile == null)
+        if (logConfigFile == null )
         {
             logConfigFile = FileUtils.findFileByName(folder, "log4j2.xml", true);
         }
 
-        String logConfigPath = logConfigFile != null ? logConfigFile.getAbsolutePath() : NULL_CONFIG_FILE;
+        String logConfigPath = logConfigFile != null && !logConfigFile.isDirectory() ? logConfigFile.getAbsolutePath() : NULL_CONFIG_FILE;
```
Comment (marcosnc) [inline core/src/test/java/org/mule/util/TestsLogConfigurationHelper.java]: Use isFile() instead.

**Pull request closed without merging**